OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -669,7 +669,10 @@ void IDeathActorInitModifier.ModifyDeathActorInit(Actor self, TypeDictionary ini
 
 		void INotifyBecomingIdle.OnBecomingIdle(Actor self)
 		{
-			if (TopLeft.Layer == 0)
```

Comment:
If we keep the `self.Location.Layer == 0` check by itself at the start we can avoid the `TraitsImplementing` lookup in 99.9% of cases.